tests/v2/validation/charts/backup_restore/backup_restore.go

```diff
@@ -0,0 +1,332 @@
+	}
+
+	config.LoadConfig(charts.BackupRestoreConfigurationFileKey, backupRestoreConfig)
+
+	chartInstallOptions := charts.InstallOptions{
+		Cluster:   cluster,
+		Version:   latestBackupVersion,
+		ProjectID: project.ID,
+	}
+	chartFeatureOptions := &charts.RancherBackupOpts{
+		VolumeName:                backupRestoreConfig.VolumeName,
+		BucketName:                backupRestoreConfig.S3BucketName,
+		CredentialSecretName:      secretName,
+		CredentialSecretNamespace: backupRestoreConfig.CredentialSecretNamespace,
+		Enabled:                   backupEnabled,
+		Endpoint:                  backupRestoreConfig.S3Endpoint,
+		Folder:                    backupRestoreConfig.S3FolderName,
+		Region:                    backupRestoreConfig.S3Region,
+	}
+
+	_, err = createOpaqueS3Secret(client.Steve)
+	if err != nil {
+		return err
+	}
+
+	logrus.Info("Installing backup chart")
+	err = charts.InstallRancherBackupChart(client, &chartInstallOptions, chartFeatureOptions, true)
+	if err != nil {
+		return err
+	}
+
+	logrus.Info("Waiting for backup chart deployments to have expected number of available replicas")
+	err = shepCharts.WatchAndWaitDeployments(client, project.ClusterID, "cattle-resources-system", metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+func createOpaqueS3Secret(steveClient *v1.Client) (string, error) {
+	config.LoadConfig(charts.BackupRestoreConfigurationFileKey, backupRestoreConfig)
+
+	logrus.Infof("Creating an opaque secret with name: %v", secretName)
+	secretTemplate := secrets.NewSecretTemplate(secretName, backupRestoreConfig.CredentialSecretNamespace, map[string][]byte{"accessKey": []byte(backupRestoreConfig.AccessKey), "secretKey": []byte(backupRestoreConfig.SecretKey)}, corev1.SecretTypeOpaque)
+	createdSecret, err := steveClient.SteveType(secrets.SecretSteveType).Create(secretTemplate)
+
+	return createdSecret.Name, err
+}
+
+func createRancherResources(client *rancher.Client, clusterID string, context string) ([]*management.User, []*management.Project, []*management.RoleTemplate, error) {
+	userList := []*management.User{}
+	projList := []*management.Project{}
+	roleList := []*management.RoleTemplate{}
+
+	for i := 0; i < resourceCount; i++ {
+		u, err := users.CreateUserWithRole(client, users.UserConfig(), "user")
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		userList = append(userList, u)
+
+		p, _, err := projects.CreateProjectAndNamespace(client, clusterID)
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		projList = append(projList, p)
+
+		rt, err := client.Management.RoleTemplate.Create(
+			&management.RoleTemplate{
+				Context: context,
+				Name:    roleName,
+				Rules:   rules,
+			})
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		roleList = append(roleList, rt)
+	}
+
+	return userList, projList, roleList, nil
+}
+
+func createBackup(client *rancher.Client, bucket string) (*v1.SteveAPIObject, string, error) {
```

Comment:
since this also validates the backup is complete, I would suggest giving this helper a more clear name like `createAndValidateBackup`